woodwork/deserialize.py

```diff
@@ -121,18 +122,21 @@ def _typing_information_to_woodwork_table(table_typing_info, **kwargs):
         use_standard_tags=table_typing_info.get('use_standard_tags'),
         table_metadata=table_typing_info.get('table_metadata'),
         column_metadata=column_metadata,
-        column_descriptions=column_descriptions)
+        column_descriptions=column_descriptions,
+        validate=validate)
 
     return dataframe
 
 
-def read_woodwork_table(path, profile_name=None, **kwargs):
+def read_woodwork_table(path, profile_name=None, validate=False, **kwargs):
     """Read Woodwork table from disk, S3 path, or URL.
 
         Args:
             path (str): Directory on disk, S3 path, or URL to read `woodwork_typing_info.json`.
             profile_name (str, bool): The AWS profile specified to write to S3. Will default to None and search for AWS credentials.
                 Set to False to use an anonymous profile.
+            validate (bool, optional): Whether parameter and data validation should occur when initializing Woodwork dataframe
+                during deserialization. Defaults to False.
```

Comment:
I'm not sure when a user might run into a situation where they'd have changed the serialized data in a way where they'd want to validate, but might be worth a warning/message to the user that if they're not sure that their data is still valid, they should set `validate=True`

Reply:
Yeah, that's a fair point. I was not thinking that users would change the serialized data, but I'm sure that's something that could happen. I'll add that to the docstring.

Reply:
Updated!